base-server/index.js

```diff
@@ -373,6 +423,16 @@ export class BaseServer {
           }
         })
     )
+    this.unbind.push(() => {
+      let queues = [...this.queues.values()]
+      let promises = queues.map(
```

Comment:
Let’s combine these three steps. In this specific case it will simplify readability (because we need `Promise.allSettled` to see first to understand what we are doing.

```js
return Promise.allSettled(
  [...this.queues.values()].map(queue => {
    return new Promise(resolve => {
      queue.drain = resolve
    })
  })
)
```